js/wp-seo-shortcode-plugin.js

```diff
@@ -0,0 +1,281 @@
+		if ( typeof callback === 'function' ) {
+			callback();
+		}
+	}
+	else if ( this.registerTime < 1001 ) {
+		this.registerTime += 100;
+		setTimeout( function() {
+			yoastShortcodePlugin.register( callback );
+		}, 100 );
+	}
+	else {
+		console.error('Failed to register shortcode plugin with YoastSEO. YoastSEO is not available.');
+	}
+};
+
+/**
+ * Declares ready with YoastSEO.
+ */
+YoastShortcodePlugin.prototype.declareReady = function() {
+	YoastSEO.app.plugins.ready( 'YoastShortcodePlugin' );
+	yoastShortcodePlugin.registerModifications();
+};
+
+/**
+ * Declares reloaded with YoastSEO.
+ */
+YoastShortcodePlugin.prototype.declareReloaded = function() {
+	YoastSEO.app.plugins.reloaded( 'YoastShortcodePlugin' );
+};
+
+/**
+ * Registers the modifications for the content in which we want to replace shortcodes.
+ */
+YoastShortcodePlugin.prototype.registerModifications = function() {
+	YoastSEO.app.plugins.registerModification( 'title', this.replaceShortcodes, 'YoastShortcodePlugin' );
+	YoastSEO.app.plugins.registerModification( 'content', this.replaceShortcodes, 'YoastShortcodePlugin' );
+};
+
+/**
+ * The callback used to replace the shortcodes.
+ *
+ * @param {string} data
+ * @returns {string}
+ */
+YoastShortcodePlugin.prototype.replaceShortcodes = function( data ) {
+	var parsedShortcodes = yoastShortcodePlugin.parsedShortcodes;
+
+	if ( typeof data === 'string' && parsedShortcodes.length > 0 ) {
+		for ( var i in parsedShortcodes ) {
+			data = data.replace( parsedShortcodes[ i ].shortcode, parsedShortcodes[ i ].output );
+		}
+	}
+	return data;
+};
+
+/* DATA SOURCING */
+
+/**
+ * Get data from inputfields and store them in an analyzerData object. This object will be used to fill
+ * the analyzer and the snippetpreview
+ */
+YoastShortcodePlugin.prototype.loadShortcodes = function() {
+	var title = document.getElementById( 'title' ).value;
+	var SEOTitle = document.getElementById( 'yoast_wpseo_title' ).value;
+	var content = this.getContentTinyMCE();
+	var meta = document.getElementById( 'yoast_wpseo_metadesc' ).value;
+
+	this.checkShortcodes( title );
+	this.checkShortcodes( SEOTitle );
+	this.checkShortcodes( content );
+	this.checkShortcodes( meta );
+
+	this.parseShortcodes( this.declareReady );
+};
+
+/**
+ * Bind elements to be able to reload the dataset if shortcodes get added.
+ */
+YoastShortcodePlugin.prototype.bindElementEvents = function() {
+	var elems = [ 'title', 'snippet_title', 'snippet_meta', 'content' ];
+
+	for ( var i = 0; i < elems.length; i++ ) {
+		document.getElementById( elems[ i ] ).addEventListener( 'keydown', yoastShortcodePlugin.inputEventCallback );
+		document.getElementById( elems[ i ] ).addEventListener( 'input', yoastShortcodePlugin.inputEventCallback );
+	}
+};
+
+/**
+ * Callback function for events on input elements. Beware! This function is always called within the scope of the element!
+ *
+ * @param {object} ev The event object
```

Comment:
The JSDoc can be more specific by being an `Event`. Maybe even more specific by being an `InputEvent` or `KeyboardEvent` (I think you'll have to find the correct one, `Event` works as well though)